packages/sftp-client/index.js

```diff
@@ -21,35 +20,33 @@ class SftpClient {
     if (config.username) this.clientOptions.username = config.username;
     if (config.password) this.clientOptions.password = config.password;
     if (config.privateKey) this.clientOptions.privateKey = config.privateKey;
+
+    this.sftpClient = new Client();
   }
 
-  /**
-   * @private
-   */
   async connect() {
     if (this.connected) return;
 
-    await new Promise((resolve, reject) => {
-      this.client = new Client();
+    await this.sftpClient.connect(this.clientOptions);
 
-      this.client.on('ready', () => {
-        this.client.sftp((err, sftp) => {
-          if (err) return reject(err);
-          this.sftp = sftp;
-          this.connected = true;
-          return resolve();
-        });
-      });
+    this.connected = true;
+  }
 
-      this.client.on('error', reject);
+  async end() {
+    if (this.connected) {
+      this.connected = false;
 
-      this.client.connect(this.clientOptions);
-    });
+      await this.sftpClient.end();
```

Comment:
We probably want to do the await before setting `this.connected` = false

Reply:
0cfb14418